src/Materials/Textures/Packer/packer.ts

```diff
@@ -0,0 +1,761 @@
+	 */
+    uvsIn?: string; // | string[];
+
+    /**
+	 * the UV output targets, as a single value for all meshes or an array of values that matches the mesh count.  Defaults to VertexBuffer.UVKind
+	 */
+    uvsOut?: string; // | string[];
+
+    /**
+	 * number representing the layout style. Defaults to LAYOUT_STRIP
+	 */
+    layout?: number;
+
+    /**
+	 * number of columns if using custom column count layout(2).  This defaults to 4.
+	 */
+    colnum?: number;
+
+    /**
+	 * flag to update the input meshes to the new packed texture after compilation. Defaults to true.
+	 */
+    updateInputMeshes?: boolean;
+
+    /**
+	* boolean flag to dispose all the source textures.  Defaults to true.
+	*/
+    disposeSources?: boolean;
+
+    /**
+	 * Fills the blank cells in a set to the customFillColor.  Defaults to true.
+	 */
+    fillBlanks?: boolean;
+
+    /**
+	 * string value representing the context fill style color.  Defaults to 'black'.
+	 */
+    customFillColor?: string;
+
+    /**
+	 * Width and Height Value of each Frame in the TexturePacker Sets
+	 */
+    frameSize?: number;
+
+    /**
+	 * Ratio of the value to add padding wise to each cell.  Defaults to 0.0115
+	 */
+    paddingRatio?: number;
+
+    /**
+	 * Number that declares the fill method for the padding gutter.
+	 */
+    paddingMode?: number;
+
+    /**
+	 * If in SUBUV_COLOR padding mode what color to use.
+	 */
+    paddingColor?: Color3 | Color4;
+
+}
+
+/**
+ * This is a support class that generates a series of packed texture sets.
+ * @see #TODO ADD THIS
+ */
+export class TexturePacker{
+
+    /** Packer Layout Constant 0 */
+    public static readonly LAYOUT_STRIP = Constants.LAYOUT_STRIP;
+    /** Packer Layout Constant 1 */
+    public static readonly LAYOUT_POWER2 = Constants.LAYOUT_POWER2;
+    /** Packer Layout Constant 2 */
+    public static readonly LAYOUT_COLNUM = Constants.LAYOUT_COLNUM;
+
+    /** Packer Layout Constant 0 */
+    public static readonly SUBUV_WRAP = Constants.SUBUV_WRAP;
+    /** Packer Layout Constant 1 */
+    public static readonly SUBUV_EXTEND = Constants.SUBUV_EXTEND;
+    /** Packer Layout Constant 2 */
+    public static readonly SUBUV_COLOR = Constants.SUBUV_COLOR;
+
+    /** The Name of the Texture Package */
+    public name: string;
+
+    /** The scene scope of the TexturePacker */
+    public scene: Scene;
+
+    /** The Meshes to target */
+    public meshes: AbstractMesh[];
+
+    /** Arguments passed with the Constructor */
+    public options: ITexturePackerOptions;
+
+    /** The promise that is started upon initialization */
+    public promise: Promise< TexturePacker | string >;
+
+    /** The Container object for the channel sets that are generated */
+    public sets: object;
+
+    /** The Container array for the frames that are generated */
+    public frames: TexturePackerFrame[];
+
+    /** The List of textures to purge from memory after compilation */
+    private _disposeList: (Texture | DynamicTexture)[];

```

Comment:
I do not see where this is used

Reply:
hmmm I guess I ended up not using it, in my cleanup method.

Let me get rid of it and change that to just a expected count, because that is needed